src/MoBi.Presentation/Nodes/TreeNodeFactory.cs

```diff
@@ -107,6 +123,38 @@ public ITreeNode CreateFor(SimulationConfiguration simulationConfiguration)
          return buildConfigNode;
       }
 
+      public ITreeNode CreateFor(ModuleConfiguration moduleConfiguration)
+      {
+         var moduleConfigurationNode = new ModuleConfigurationNode(moduleConfiguration).WithIcon(ApplicationIcons.Module);
+         var module = moduleConfiguration.Module;
+
+         addModuleBuildingBlocks(moduleConfigurationNode, module);
+         
+         if(moduleConfiguration.SelectedMoleculeStartValues != null)
+            createAndAddNodeUnder(moduleConfigurationNode, moduleConfiguration.SelectedMoleculeStartValues);
```

Comment:
whata about having the `createAndAddNodeUnder` method check that the building block is not null  instead of checking all the time?  you are doing it for `addBuildingBlockNodeUnder` so it would be more cohesive